playbooks/CVE-2015-3456.yml

```diff
@@ -1,25 +1,72 @@
 ---
 - hosts: compute
   tasks:
-  - name: update apt cache
-    apt: update_cache=yes cache_valid_time=3600
+  - name: Install key for Ubuntu Cloud Archive
+    apt: pkg=ubuntu-cloud-keyring update_cache=yes
+    when: ansible_distribution_version == "12.04"
 
-  - name: Patch CVE-2015-3456 on precise
-    apt: pkg={{ item }} state=latest
-    with_items:
-    - qemu
-    - qemu-kvm
-    notify: restart qemu-kvm
+  - name: Ubuntu Cloud Archive Repository
+    apt_repository: repo='deb http://ubuntu-cloud.archive.canonical.com/ubuntu precise-updates/icehouse main'
+                    update_cache=yes
     when: ansible_distribution_version == "12.04"
 
-  - name: Patch CVE-2015-3456 on trusty
-    apt: pkg={{ item }} state=present
+  - name: Upgrade QEMU
+    apt: pkg={{ item }}
     with_items:
-    - qemu=2.0.0+dfsg-2ubuntu1.11
-    - qemu-kvm=2.0.0+dfsg-2ubuntu1.11
-    notify: restart qemu-kvm
+    - qemu-kvm=2.0.0+dfsg-2ubuntu1.11~cloud0
+    - librbd1
+    - python-libvirt
+    when: ansible_distribution_version == "12.04"
+
+  - name: Upgrade QEMU
+    apt: pkg=qemu-kvm=2.0.0+dfsg-2ubuntu1.11 update_cache=yes
+    when: ansible_distribution_version == "14.04"
+
+  - name: Upgrade kernel
+    apt: pkg=linux-generic-lts-trusty
+    when: ansible_distribution_version == "12.04"
+
+  - name: Upgrade kernel
+    apt: pkg=linux-generic
     when: ansible_distribution_version == "14.04"
 
-  handlers:
-  - name: restart qemu-kvm
-    service: name=qemu-kvm state=restarted
+- hosts: compute:!db
+  tasks:
+  - name: Reboot
+    command: shutdown -r now
+    async: 0
+    poll: 0
+    ignore_errors: true
```

Comment:
failed_when: false is the better option, saves red output. But doesn't matter for this task.